src/main/java/com/faforever/client/vault/search/SearchController.java

```diff
@@ -429,7 +430,14 @@ public void setRootType(Class<?> rootType) {
   }
 
   public void setSearchButtonDisabledCondition(BooleanBinding inSearchableState) {
-    searchButton.disableProperty().bind(queryTextField.textProperty().isEmpty().or(inSearchableState.not()));
+    Optional<Property> firstSortProperty = sortPropertyComboBox.getItems().stream().findFirst();
+    BooleanBinding isSortChanged = firstSortProperty.isEmpty()
+        ? Bindings.selectBoolean(false)
+        : sortPropertyComboBox.valueProperty().isNotEqualTo(firstSortProperty.get());
+    searchButton.disableProperty().bind(
+        queryTextField.textProperty().isEmpty().and(isSortChanged.not())
+        .or(inSearchableState.not())
+    );
```

Comment:
I don't understand why this change is necessary. If this is just checking if the sort property is populated that is easier done by setting the combo box to not have editable or null values which I believe is already done.

Additionally right now this is only binding on the first item in the combobox rather than the selected item of the combo box so will have unintended behavior. And you will never be able to sort by the first property in the list.

As you can see in the image below it disables the search when sort is the first value which shouldn't happen. I don't think this needs to be changed at all.
![image](https://user-images.githubusercontent.com/66929319/174418454-f2fa0343-91d7-442f-8629-5934667c872f.png)

Reply:
the reason I did this is so that you can do a search with no filters and a different sort by.

Reply:
Then it would be better to just explicitly allow for searching with no filters than do this as it doesn't make sense why you just can't search with one of the sort options

Reply:
that makes sense